src/python/test/sugar/test_xfer.py

```diff
@@ -0,0 +1,291 @@
+from __future__ import print_function, unicode_literals, division, absolute_import
+import contextlib
+import gzip
+import os
+import shutil
+import tempfile
+import unittest
+from uuid import uuid4
+
+import dxpy
+from dxpy.sugar import xfer
+from dxpy.sugar.proc import run
+
+
+def random_name(fmt_str=None):
+    rndstr = str(uuid4())
+    if fmt_str:
+        return fmt_str.format(rndstr)
+    else:
+        return rndstr
+
+
+def make_random_files(n, fmt_str=None):
+    filenames = []
+    for i in range(n):
+        fname = random_name(fmt_str)
+        filenames.append(fname)
+        with open(fname, "wt") as out:
+            out.write("test{}".format(i))
+    return filenames
+
+
+@contextlib.contextmanager
+def temp_dir(*args, **kwargs):
+    dname = tempfile.mkdtemp(*args, **kwargs)
+    try:
+        yield dname
+    finally:
+        shutil.rmtree(dname)
+
+
+@contextlib.contextmanager
+def isolated_dir():
+    with temp_dir() as d:
+        curdir = os.getcwd()
+        os.chdir(d)
+        try:
+            yield d
+        finally:
+            os.chdir(curdir)
```

Comment:
The context managers here should go in the context_managers module instead of here. I think both of these are handled by the cd() context manager already

Reply:
Agreed - it's just a dependency issue. I can't use them until they are committed to the master branch.

Reply:
Oh, just saw Steve's comment. I will rebase to SCI-1321_dx_sugar.